### ISSUE TITLE
ArrayIndexOutOfBoundsException reading from stream

### ISSUE THREAD

**Issue by petershu** [square/okhttp#1540]
Our crash logger is logging a lot of these ArrayIndexOutOfBoundsExceptions when trying to read GZIP data from the response body's byte stream:

```
java.lang.ArrayIndexOutOfBoundsException: size=0 offset=0 byteCount=1
	at okio.Util.checkOffsetAndCount(Util.java:29)
	at okio.Buffer.getByte(Buffer.java:294)
	at okio.RealBufferedSource.readHexadecimalUnsignedLong(RealBufferedSource.java:269)
	at com.squareup.okhttp.internal.http.HttpConnection$ChunkedSource.readChunkSize(HttpConnection.java:451)
	at com.squareup.okhttp.internal.http.HttpConnection$ChunkedSource.read(HttpConnection.java:432)
	at okio.RealBufferedSource$1.read(RealBufferedSource.java:338)
	at java.io.InputStream.read(InputStream.java:162)
	at java.util.zip.InflaterInputStream.fill(InflaterInputStream.java:194)
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:153)
	at java.util.zip.GZIPInputStream.read(GZIPInputStream.java:163)
	at java.io.FilterInputStream.read(FilterInputStream.java:118)
```

Any suggestions?

**Comment by swankjesse** [comment]
Looks like a bug in our code. Will fix.

**Comment by JakeWharton** [comment]
Fix is merged into Okio and will be in the next OkHttp release.

**Comment by cocoahero** [comment]
Any chance this patch can be released as a 2.3.x bug fix? We can catch the `ArrayIndexOutOfBoundsException` to prevent our apps from crashing, but it still means requests will never successfully complete, which is almost as bad.

**Comment by swankjesse** [comment]
I'm hoping to cut 2.4 in 5 weeks or fewer. If you'd prefer sooner, please use a snapshot! (Sorry!)
